src/adapter/src/catalog.rs

```diff
@@ -3843,6 +3844,7 @@ impl<S: Append> Catalog<S> {
     #[tracing::instrument(level = "debug", skip_all)]
     pub async fn transact<F, R>(
         &mut self,
+        oracle_write_ts: Option<mz_repr::Timestamp>,
```

Comment:
This `Option` seems like asking for trouble. This cannot be `None` unless the provided ops don't write to the audit log. That seems very hard to assess as the caller, and also the condition is not documented.

Reply:
I pushed up a commit that makes this change. LMK what you think.